### PR TITLE
Add tests for setting cluster version using v2 request

### PR DESCRIPTION
**Pull request by serathius**
Split off https://github.com/etcd-io/etcd/pull/16989

### CODE REVIEW

**Comment by serathius** [comment]
cc @ahrtr

File: server/etcdserver/server_test.go [hunk 1]
```diff
@@ -401,6 +405,33 @@ func TestApplyRequestOnAdminMemberAttributes(t *testing.T) {
...
+		v2store: mockstore.NewRecorder(),
+		cluster: cl,
+	}
+	srv.applyV2 = NewApplierV2(srv.lg, srv.v2store, srv.cluster)
+
+	req := pb.Request{
+		Method: "PUT",
+		ID:     1,
+		Path:   membership.StoreClusterVersionKey(),
+		Val:    `3.6.0`,
```
Comment (ahrtr) [inline server/etcdserver/server_test.go:427]:
```suggestion
		Val:    `3.5.0`,
```

File: server/etcdserver/server_test.go [hunk 2]
```diff
@@ -401,6 +405,33 @@ func TestApplyRequestOnAdminMemberAttributes(t *testing.T) {
...
+	}
+	srv.applyV2Request((*RequestV2)(&req), membership.ApplyBoth)
+	if g := cl.Version(); !reflect.DeepEqual(*g, version.V3_6) {
```
Comment (ahrtr) [inline server/etcdserver/server_test.go:430]:
```suggestion
	if g := cl.Version(); !reflect.DeepEqual(*g, version.V3_5) {
```

File: server/etcdserver/server_test.go (hunk 1 above)
Comment (ahrtr) [reply server/etcdserver/server_test.go:427]: nit: Only 3.5 will use v2 PUT request

**Review by ahrtr — APPROVED** [review]
lgtm with minor comments.

File: server/etcdserver/server_test.go (hunk 1 above)
Comment (serathius) [reply server/etcdserver/server_test.go]: Done

**Pull request merged**